x-pack/legacy/plugins/ml/public/application/routing/routes/timeseriesexplorer.tsx

```diff
@@ -282,6 +282,11 @@ export const TimeSeriesExplorerUrlStateManager: FC<TimeSeriesExplorerUrlStateMan
     return null;
   }
 
+  const zoomProp: AppStateZoom | undefined =
+    typeof selectedForecastId === 'string' && selectedForecastIdProp === undefined
```

Comment:
does zoom prop always depend on the forecast id?

Reply:
This is related to the comment above. The combination of `selectedForecastId` and `selectedForecastIdProp` allows us to decide wheter to pass on `zoomProp` as `undefined`. If it's `undefined`, the `timeseriesexplorer.js` will recalculate the zoom which in turn moves it to the desired location to span half the forecast and have the existing section.